client/src/components/modals/ComponentModal.tsx

```diff
@@ -0,0 +1,278 @@
+import { CloudUpload, Info } from '@mui/icons-material';
+import CloseIcon from '@mui/icons-material/Close';
+import {
+	Alert,
+	Box,
+	Button,
+	Chip,
+	Dialog,
+	DialogActions,
+	DialogContent,
+	DialogTitle,
+	FormControl,
+	IconButton,
+	InputLabel,
+	MenuItem,
+	OutlinedInput,
+	Select,
+	Stack,
+	TextField,
+	Tooltip
+} from '@mui/material';
+import axios from 'axios';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { parseConfigFileTextToJson, readJsonConfigFile } from 'typescript';
+import { dataConfigParser } from '../../utils/data-parser';
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+	PaperProps: {
+		style: {
+			maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+			width: 250
+		}
+	}
+};
+
+interface ComponentModalProps {
+	rocketProfileId: 0;
+	isOpen: boolean;
+	onSave: () => void;
+	onClose: () => void;
+}
+
+interface dataConfigStructure {
+	config: {};
+	data: { results: { [key: string]: any } };
+	headers: {};
+	request: {};
+	status: number;
+	statusText: string;
+}
+
+const ComponentModal = (props: ComponentModalProps) => {
+	const [name, setName] = useState<string>('');
+	const [details, setDetails] = useState<string>('');
+	const [sourceTypes, setSourceTypes] = useState<string[]>([]);
+	const [editMode, setEditMode] = useState<boolean>(false);
+	const [errorBar, setErrorBar] = useState({
+		show: false,
+		message: 'Error Occured'
+	});
+	const [configFile, setConfigFile] = useState<File | null>(null);
+	const parsedConfigFile = useRef<{ [key: string]: Object }>({});
+
+	const handleChange = (e: any, setState: Function) => {
+		if (!editMode) {
+			setEditMode(true);
+		}
+		setState(e.target.value);
+	};
+	const save = async (): Promise<boolean> => {
+		const response: dataConfigStructure = await axios.post(
+			`http://127.0.0.1:9090/DataConfig`,
+			parsedConfigFile.current
+		);
+		let dataConfigId;
```

Comment:
```suggestion
		let dataConfigId: string;
```

I was getting a type error without this